client/blocks/reader-post-card/compact.jsx

```diff
@@ -70,13 +93,25 @@ const CompactPost = ( {
 				{ post.canonical_media && (
 					<div className="reader-post-card__post-media">
 						{ ! isSmallScreen && hasExcerpt && (
-							<ReaderPostEllipsisMenu
-								site={ site }
-								teams={ teams }
-								post={ post }
-								showFollow
-								openSuggestedFollows={ openSuggestedFollows }
-							/>
+							<div className="reader-post-card__post-options">
+								{ isDiscover && (
+									<ReaderFollowButton
+										tagName="div"
+										siteUrl={ post.feed_URL || post.site_URL }
+										followSource={ READER_DISCOVER }
+										iconSize={ 20 }
+										followLabel={ translate( 'Subscribe' ) }
+										followingLabel={ translate( 'Unsubscribe' ) }
+									/>
+								) }
+								<ReaderPostEllipsisMenu
+									site={ site }
+									teams={ teams }
+									post={ post }
+									showFollow
+									openSuggestedFollows={ openSuggestedFollows }
+								/>
+							</div>
```

Comment:
We are duplicating the code here which we should avoid.

Reply:
I've consolidated the duplicated code into a common component 👍